src/pykeen/pipeline.py

```diff
@@ -699,6 +699,34 @@ def pipeline_from_config(
     )
 
 
+def save_pipeline_checkpoint_helper_file(path: str, random_seed: int) -> None:
+    """Save the pipeline checkpoint helper file.
+
+    :param path:
+        Save the state of the pipeline.
+    :param random_seed:
+        The random_seed that was used for the pipeline.
+    """
+    torch.save(
+        {
+            'random_seed': random_seed,
```

Comment:
why is the random seed relevant? so it doesn't get overwritten?

Reply:
When we run a pipeline experiment, we save the random_seed as well, since it is important for all random parts. When the user starts a pipeline without providing a random seed, a random_seed is generated say `random_seed=1`. When the users pauses the training and resumes it later on by starting the same pipeline again, a new random seed would be generated, say `random_seed=2`. When the experiments finishes, it would state that the experiment was run with `random_seed=2`.
This problem is remedied by storing and re-using the random seed from the pipeline helper file.

Reply:
There is still a difference in starting a run with a fixed random seed and letting it run until the end, and stopping in between and restarting from a checkpoint: When restarting from a checkpoint, the last random state does not get restored. Thus, all random parts, e.g. negative sampling, will yield different results than they had done when running through.

I am not sure whether we can really solve this problem though, or if it would be better to guarantee it for running an experiments, but not when restoring from checkpoints.

Reply:
it might be possible to acquire the current random state from PyTorch... but like once you're at this point, you're probably not really worried about **re**producibility so much as reproducibility for the first time lol

Reply:
So the `pytorch.Generator()` object could address this problem, but we'd have to do a pretty major refactoring to use our own instance of one of those.

Ref: https://github.com/pytorch/pytorch/issues/17079

Reply:
This should be fixed with https://github.com/pykeen/pykeen/pull/123/commits/fc9a5a9fb01e6a9edd5db92dbc498b0f27fe8b17

Reply:
That's awesome!!